apps/ui/src/views/Proposal/Overview.vue

```diff
@@ -114,6 +119,34 @@ async function handleCancelClick() {
     cancelling.value = false;
   }
 }
+
+async function handleAiSummaryClick() {
+  if (!aiSummaryBody.value) {
+    try {
+      aiSummaryLoading.value = true;
+      const response = await fetch(`https://sh5.co/api/ai/summary/${props.proposal.id}`, {
+        method: 'POST'
+      });
+      const data = await response.json();
+
+      if (data.error) {
+        throw new Error(data.error.message);
+      }
+
+      aiSummaryBody.value = data.result;
+    } catch (e) {
+      aiSummaryError.value = e;
```

Comment:
Looks like we are never clearing this value so on retry it will always be in error state and show notification I think.

I don't think it even needs to be a ref as we only use it here. What do you think about writing it this way?

```js
async function handleAiSummaryClick() {
  if (aiSummaryBody.value) return;

  try {
    aiSummaryLoading.value = true;
    const response = await fetch(`https://sh5.co/api/ai/summary/${props.proposal.id}`, {
      method: 'POST'
    });
    const data = await response.json();

    if (data.error) {
      throw new Error(data.error.message);
    }

    aiSummaryBody.value = data.result;
    aiSummaryOpen.value = !aiSummaryOpen.value;
  } catch (e) {
    uiStore.addNotification('error', 'There was an error fetching the AI summary.');
  } finally {
    aiSummaryLoading.value = false;
  }
}
```

Reply:
You're correct about the error ref. `if (aiSummaryBody.value) return;` however can not be used since this function also toggle the `open` state below

Reply:
Got it, I think it's a bit weird to read because of that. I came up with this, to have exit-early approach instead of skipping 90% of function to reach side effects.
```js
async function handleAiSummaryClick() {
  if (aiSummaryBody.value) {
    aiSummaryOpen.value = !aiSummaryOpen.value;
    return;
  }

  try {
    aiSummaryLoading.value = true;
    const response = await fetch(`https://sh5.co/api/ai/summary/${props.proposal.id}`, {
      method: 'POST'
    });
    const data = await response.json();

    if (data.error) {
      throw new Error(data.error.message);
    }

    aiSummaryBody.value = data.result;
    aiSummaryOpen.value = true;
  } catch (e) {
    uiStore.addNotification('error', 'There was an error fetching the AI summary.');
  } finally {
    aiSummaryLoading.value = false;
  }
}
```

If you don't like it we could keep it as-is as well as I don't think it makes a huge difference.

Reply:
This duplicates the opening logic, but also works, refactored